plasmapy/diagnostics/thomson.py

```diff
@@ -204,114 +468,427 @@ def spectral_density(
-            2
-            * np.sqrt(np.pi)
-            / k
-            / vTe[m]
-            * np.power(np.abs(1 - np.sum(chiE, axis=0) / epsilon), 2)
-            * np.exp(-xe[m, :] ** 2)
+
+def _spectral_density_model(wavelengths, settings=None, **params):
+    """
+    lmfit Model function for fitting Thomson spectra
+
+    For descriptions of arguments, see the `thomson_model` function.
+
+    """
+
+    # LOAD FROM SETTINGS
+    ion_z = settings["ion_z"]
+    ion_mass = settings["ion_mass"]
+    probe_vec = settings["probe_vec"]
+    scatter_vec = settings["scatter_vec"]
+    electron_vdir = settings["electron_vdir"]
+    ion_vdir = settings["ion_vdir"]
+    probe_wavelength = settings["probe_wavelength"]
+    instr_func_arr = settings["instr_func_arr"]
+
+    # LOAD FROM PARAMS
+    n = params["n"]
+    Te = _params_to_array(params, "Te")
+    Ti = _params_to_array(params, "Ti")
+    efract = _params_to_array(params, "efract")
+    ifract = _params_to_array(params, "ifract")
+
+    electron_speed = _params_to_array(params, "electron_speed")
+    ion_speed = _params_to_array(params, "ion_speed")
+
+    electron_vel = electron_speed[:, np.newaxis] * electron_vdir
+    ion_vel = ion_speed[:, np.newaxis] * ion_vdir
+
+    # Convert temperatures from eV to Kelvin (required by fast_spectral_density)
+    Te *= 11605
+    Ti *= 11605
+
+    alpha, model_Skw = spectral_density_lite(
+        wavelengths,
+        probe_wavelength,
+        n,
+        Te,
+        Ti,
+        efract=efract,
+        ifract=ifract,
+        ion_z=ion_z,
+        ion_mass=ion_mass,
+        electron_vel=electron_vel,
+        ion_vel=ion_vel,
+        probe_vec=probe_vec,
+        scatter_vec=scatter_vec,
+        instr_func_arr=instr_func_arr,
+    )
+
+    model_Skw *= 1 / np.max(model_Skw)
+
+    return model_Skw
+
+
+def spectral_density_model(wavelengths, settings, params):
+    """
+    Returns a `lmfit.Model` function for Thomson spectral density function
+
+
+    Parameters
+    ----------
+
+
+    wavelengths : np.ndarray
+        Wavelength array, in meters.
+
+
+    settings : dict
+        A dictionary of non-variable inputs to the spectral density function
+        which must include the following:
+
+            - probe_wavelength: Probe wavelength in meters
+            - probe_vec : (3,) unit vector in the probe direction
+            - scatter_vec: (3,) unit vector in the scattering direction
+            - ion_species : list of Particle strings describing each ion species
+
+        and may contain the following optional variables
+            - electron_vdir : (e#, 3) array of electron velocity unit vectors
+            - ion_vdir : (e#, 3) array of ion velocity unit vectors
+            - instr_func : A function that takes a wavelength u.Quantity array
+                        and returns a spectrometer insturment function as an
+                        `numpy.ndarray`.
+
+        These quantities cannot be varied during the fit.
```

Comment:
I was wondering if this would be a good place to create a [parameter object](https://refactoring.guru/introduce-parameter-object).  We could create a class named something like `ThomsonModelSettings`, and have each of the things that go into the `dict` be attributes of that class instead, each with their own docstring.  That approach would also provide a place for validation & processing of the parameters if needed that would be outside of this function.  That wouldn't necessarily need to be in this pull request though.

Reply:
That's an idea: we could also split the validation into a separate private function. I kind of like having the `parameters` interface exposed so the users can read the `lmfit` docs and use all the various tools there without me passing stuff through.